### ISSUE TITLE
[QA] new deployed feature causes 3-dots-menue to have random contents

### ISSUE THREAD

**Issue by jnweiger** [owncloud/client#9954]
### Pre-submission Checks

- [X] I [checked for similar issues](https://github.com/owncloud/client/issues?q=), but could not find any. I also checked the closed issues. I could not contribute additional information to any existing issue.
- [X] I will take the time to fill in all the required fields. I know that the bug report may be dismissed otherwise due to lack of information.

### Describe the QA issue

Seen with 2.10.2-rc.1 (7957) while trying to reproduce #9791 

Reproducable both on linux and win11:

* connect client to server without choosing a folder (select 'manually later...')
* add folder /svg/ as a folder sync connection
* add folder /svg/sub as folder sync connection (warning appears, but can be added)
* add folder /svg/sub as folder sync connection again (warning appears, but can be added)

Inspect the 3-dots menu. It misses Remove entries sometimes, misses VFS entries sometimes, and has extra 'chosse what to sync ' entry sometimes.

![image](https://user-images.githubusercontent.com/1108546/180803190-3fd5e282-595f-4f52-bcab-beaa4eeb009b.png)
![image](https://user-images.githubusercontent.com/1108546/180803071-4ffaeb8f-8e0f-4506-a273-f9d8fe690e49.png)
![image](https://user-images.githubusercontent.com/1108546/180803092-621119e5-a08f-424b-a32c-1197e0310b2b.png)

Also, the three sync folders are presented inconsistently in the Windows file explorer.
One has the green checkmark overlay, but no 'ownCloud - ' prefix, 
The ownCloud sub-menue is almost always grayed out

![image](https://user-images.githubusercontent.com/1108546/180803560-98396aaa-4347-4fc3-add7-5017fdc43c2b.png)

### Steps to reproduce the issue

.

### Screenshots

.

### Expected behavior

Consistent menu entries, 
Remove option is always present!

### Actual behavior

.

**Comment by jnweiger** [comment]
Having multiply sync folder connections is not a prerequisite. I find a crippled 3-dots-menu also with only one sync folder connection:

![image](https://user-images.githubusercontent.com/1108546/180805942-9290fe48-1028-410c-b147-01ea0c6ec61e.png)

**Comment by jnweiger** [comment]
Reprouced on Linux:

* `rm -f ~/.config/testpilotcloud/testpilotcloud.cfg`
* connect testpilotcloud 2.10.2-rc.1 to two different servers.
* wait for both to sync.
* check the 3-dots menu. Chances are, one has Remove and the VFS options, the other not.
* ```grep deployed .config/testpilotcloud/testpilotcloud.cfg
0\Folders\2\deployed=false
1\Folders\1\deployed=true
```

The initialization of the `deployed` variable seems random. 
The client acts according to this variable setting.

Workaround:
* stop the client, 
* edit the config file, change all `deployed=true` to `deployed=false` 
* restart the client.
* All 3-dot-menues are now normal

**Comment by TheOneRing** [comment]
Fixed in 2.10.2 dailies
